frontend/common/src/helpers/skillUtils.ts

```diff
@@ -0,0 +1,39 @@
+import { flatMap, uniqBy } from "lodash";
+import { Skill, SkillFamily } from "../api/generated";
+
+/**
+ * Transforms an array of skills with child skill families into a tree of skill families with child skills.
+ * @param { Skill[] } skills - The collection of skills with child skill families to invert
+ * @returns { SkillFamily[] } - The new collection of skill families with child skills
+ */
+export function invertSkillTree(skills: Skill[]): SkillFamily[] {
+  const allChildSkillFamilies = flatMap(skills, "families");
```

Comment:
I wasn't sure what flatMap returns, so I tried replacing `flatMap(skills, "families")` with `flatMap(skills, s => s.families)`, which is a bit more typescript friendly. Typescript then interprets the type of the output as ` (SkillFamily | null | undefined)[]`, and the null|undefined causes issues later on. 
I know this is very unlikely to come up, but filtering out empty values will _guarantee_ consistency.
```suggestion
  const allChildSkillFamilies = flatMap(skills, (s) => s.families).filter(
    notEmpty,
  );
```

Reply:
Oh yeah, 💯 This is much better.